drivers/i2c/i2c_ll_stm32.c

```diff
@@ -103,6 +103,33 @@ static int i2c_stm32_init(struct device *dev)
 	__ASSERT_NO_MSG(clock);
 	clock_control_on(clock, (clock_control_subsys_t *) &cfg->pclken);
 
+#ifdef CONFIG_SOC_SERIES_STM32F3X
+	/*
+	 * I2C on STM32F3 series is clock independent and hence, we should
+	 * specify clock before using it.
+	 */
+
+	switch ((u32_t)cfg->i2c) {
+#ifdef CONFIG_I2C_1
+	case CONFIG_I2C_1_BASE_ADDRESS:
+		LL_RCC_SetI2CClockSource(LL_RCC_I2C1_CLKSOURCE_SYSCLK);
```

Comment:
We should take care of Clock control configuration.
If board is running on HSI, this won't work.

Reply:
I was just looking into the reference manuals. All SOCs with I2C v2 have an I2C independent clock. I will make the commit generic.

Reply:
Ok for APB1.
But looking to F334 and F303 ref manuals, in RCC chapters, clock tree diagrams show that I2Cx could be feed with HSI or SYSCLK.

Reply:
My last push checks if IS_ENABLED(CONFIG_STM32_PLL_SRC_HSI) and sets the clock accordingly.
```
if (IS_ENABLED(CONFIG_STM32_PLL_SRC_HSI)) {
	LL_RCC_SetI2CClockSource(LL_RCC_I2C1_CLKSOURCE_HSI);
} else {
	LL_RCC_SetI2CClockSource(LL_RCC_I2C1_CLKSOURCE_SYSCLK);
}
```